### DISCUSSION TITLE
System QCCSID is not set correctly

### DISCUSSION THREAD

**Discussion by cirikus** [codefori/vscode-ibmi#2356] [General]
Hello,
I don't see anything when i connect on my IBMi.

![image](https://github.com/user-attachments/assets/08f5da0d-b50a-4620-b032-5fd171f9ad7a)

I have these issues : 

![image](https://github.com/user-attachments/assets/d17f3565-8570-456c-a7b4-52bb7125ad13)

How to fix these issues please ?

Thks a lot

**Comment by sebjulliand** [comment]
Hi there,
What's your IBM i os version?

**Reply by cirikus** [reply]
Hi,
My version is V7R4M0
![image](https://github.com/user-attachments/assets/2bb8edbf-8883-40d1-b86e-b53a8959abb6)

**Reply by sebjulliand** [reply]
Ok, good.
Could you connect using the "Connect and reload settings" action? (Right click on the connection)
Then share here your entire Code for IBM i output.
![code for i output](https://github.com/codefori/vscode-ibmi/assets/11096890/88bc7363-2402-419f-b173-309358b914b9)

Thanks!

**Reply by cirikus** [reply]
Ok :)
![image](https://github.com/user-attachments/assets/e185227b-7040-4e97-938c-068e17a4e4bc)
Thks a lot

**Reply by sebjulliand** [reply]
Sorry, wrong output. Look for the code for ibm I one, in the drop-down list.
There should be a lot of text to copy and paste 😅

**Reply by cirikus** [reply]
Oh sorry 😅

.: echo "This should be the only text!"
{
    "code": 0,
    "signal": null,
    "stdout": "This should be the only text!",
    "stderr": ""
}

.: echo $HOME && cd && test -w $HOME
{
    "code": 0,
    "signal": null,
    "stdout": "/",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
liblist
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
system  "CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: [ -d "/tmp" ]
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: echo $SHELL
{
    "code": 0,
    "signal": null,
    "stdout": "/QOpenSys/pkgs/bin/bash",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
system  "CHKOBJ OBJ(QSYS/QCPTOIMPF) OBJTYPE(*DTAARA)"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
system  "CHKOBJ OBJ(QSYS/QCPFRMIMPF) OBJTYPE(*DTAARA)"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: ls -p /usr/bin/
{
    "code": 0,
    "signal": null,
    "stdout": "Rfile\najar\nappletviewer\napt\narmsrv\narmsrvconv\nattr\nbasename\nbash\ncat\ncatsplf\nchgrp\nchmod\nchown\nclrtmp\ncmp\ncompress\ncp\ncut\ndataq\ndatarea\ndate\ndb2\ndb2profc\ndb2profc_trace\ndb2profp\nddns-confgen\ndelv\ndfmadmin\ndhcpd\ndhcrelay\ndig\ndirname\ndnssec-dsfromkey\ndnssec-importkey\ndnssec-keyfromlabel\ndnssec-keygen\ndnssec-revoke\ndnssec-settime\ndnssec-signzone\ndnssec-verify\ndspmsg\necho\negrep\nenv\nexpr\nextcheck\nfgrep\nfile\nfind\ngencat\ngetjobid\ngrep\nhead\nhost\nhostname\nhttpapi\niconv\nid\nidlj\nikeyman\nipcrm\nipcs\nixlc\njar\njarsigner\njava\njavac\njavadoc\njavah\njavap\njconsole\njdb\njdmpview\njextract\nkdestroy\nkeytab\nkeytool\nkinit\nklist\nkpasswd\nksetup\nldapadd\nldapchangepwd\nldapcompare\nldapdelete\nldapdiff\nldapexop\nldapmodify\nldapmodrdn\nldapsearch\nln\nlocale\nlogger\nlogname\nls\nmkdir\nmkfifo\nmv\nnamed\nnamed-checkconf\nnamed-checkzone\nnamed-compilezone\nnamed-journalprint\nnamed-rrchecker\nnative2ascii\nnohup\nnslookup\nnsupdate\nod\norbd\npack200\npax\npolicytool\npr\nprintenv\nprofconv\nprofdb\nprofp\nps\npwd\npwdx\nqsh\nrexec\nrexx\nrm\nrmdir\nrmic\nrmid\nrmiregistry\nrndc\nrndc-confgen\nschemagen\nsed\nserialver\nservertool\nsetccsid\nsh\nsort\nsplit\nsqlj\nstrEKM\nsystem\nsysval\ntail\ntar\ntee\ntest\ntnameserv\ntouch\ntr\ntsig-keygen\nuname\nuncompress\nuniq\nunpack200\nwc\nwsgen\nwsimport\nxargs\nxjc\nzcat",
    "stderr": ""
}

/: ls -p /QOpenSys/pkgs/bin/
{
    "code": 0,
    "signal": null,
    "stdout": "2to3\n2to3-3.6\n2to3-3.9\n5250keys\n[\naddbuiltin\nalternatives\natob\nbaddbdir\nbase32\nbase64\nbasename\nbash\nbashbug\nbltest\nbtoa\nc++\nc2ph\ncat\ncc\ncertcgi\ncertutil\nchardetect\nchcon\nchgrp\nchktest\nchmod\nchown\nchroot\nchsh\ncksum\ncmsutil\ncomm\nconflict\ncorelist\ncp\ncpan\ncpp-6\ncpp-6.bin\ncrlutil\ncrmftest\ncsplit\ncurl\ncut\ncwbcopwr\ncwbmedic\ncwbnltbl\ncwbping\ncwbtrc\ncygdb\ncython\ncythonize\ndate\ndbtest\ndd\ndebuginfo-install\nderdump\ndertimetest\ndf\ndigest\ndir\ndircolors\ndirname\ndltest\ndu\necho\necperf\negrep\nenc2xs\nencguess\nencodeinttest\nenv\nexample\nexpand\nexpr\nfactor\nfalse\nfbectest\nfgrep\nfind-repos-of-install\nfipstest\nfmt\nfold\ng++\ng++-6\ng++-6.bin\ngcc\ngcc-6\ngcc-6.bin\ngcc-ar\ngcc-ar-6\ngcc-ar-6.bin\ngcc-nm\ngcc-nm-6\ngcc-nm-6.bin\ngcc-ranlib\ngcc-ranlib-6\ngcc-ranlib-6.bin\ngcov\ngcov-6\ngendiff\ngrep\ngroups\ngunzip\ngzexe\ngzip\nh2ph\nh2xs\nhead\nhostid\nhttpserv\nid\nidle3\nidle3.6\nidle3.9\ninstall\ninstmodsh\nisql\niusql\njoin\njson_pp\nkill\nlibnetcfg\nlibtool\nlibtoolize\nlink\nlistsuites\nln\nlogname\nlp5250d\nls\nmakepqg\nmangle\nmd5sum\nminigzip\nmkdir\nmkfifo\nmknod\nmktemp\nmodutil\nmultinit\nmv\nneeds-restarting\nnice\nnl\nnohup\nnonspr10\nnproc\nnumfmt\nocspclnt\nocspresp\nod\nodbc_config\nodbcinst\noidcalc\np7content\np7env\np7sign\np7verify\npackage-cleanup\npaste\npathchk\nperl\nperl5.24.1\nperlbug\nperldoc\nperlivp\nperlthanks\npiconv\npinky\npip3\npip3.9\npk11ectest\npk11gcmtest\npk11mode\npk12util\npk1sign\npkix-errcodes\npl2pm\npod2html\npod2man\npod2text\npod2usage\npodchecker\npodselect\npp\npr\nprintenv\nprintf\nprove\npstruct\nptar\nptardiff\nptargrep\nptx\npwd\npwdecrypt\npydoc\npydoc3\npydoc3.6\npydoc3.9\npython\npython2\npython2.7\npython3\npython3.6\npython3.6m\npython3.9\npython3.9-config\npyvenv-3.6\nreadlink\nrealpath\nremtest\nrepo-graph\nrepo-rss\nrepoclosure\nrepodiff\nrepomanage\nrepoquery\nreposync\nrepotrack\nrm\nrmdir\nrpm\nrpm2archive\nrpm2cpio\nrpmdb\nrpmgraph\nrpmkeys\nrpmquery\nrpmsign\nrpmspec\nrpmverify\nrsaperf\nruncon\nscs2ascii\nscs2pdf\nscs2ps\nsdrtest\nsecmodtest\nselfserv\nseq\nsha1sum\nsha224sum\nsha256sum\nsha384sum\nsha512sum\nshasum\nshlibsign\nshow-changed-rco\nshow-installed\nshred\nshuf\nsigntool\nsignver\nsleep\nslencheck\nsmtpd.py\nsort\nsplain\nsplit\nssltap\nstat\nstdbuf\nstrsclnt\nstty\nsum\nsymkeyutil\nsync\ntac\ntail\ntee\ntest\ntimeout\ntn5250\ntouch\ntox\ntr\ntrue\ntruncate\ntrust\ntsort\ntstclnt\ntty\nuname\nuncompress\nunexpand\nuniq\nunlink\nupdate-alternatives\nupdate-ca-certificates\nuptime\nurlgrabber\nusers\nvdir\nverifytree\nvfychain\nvfyserv\nvirtualenv\nwc\nwhataformatter\nwheel\nwho\nwhoami\nxml2-config\nxslt-config\nxsltproc\nxsubpp\nxt5250\nyes\nyum\nyum-builddep\nyum-complete-transaction\nyum-config-manager\nyum-debug-dump\nyum-debug-restore\nyum-groups-manager\nyumdb\nyumdownloader\nzcat\nzcmp\nzdiff\nzegrep\nzfgrep\nzforce\nzgrep\nzipdetails\nzless\nzmore\nznew",
    "stderr": ""
}

/: ls -p /QSYS.LIB/QZDFMDB2.PGM
{
    "code": 0,
    "signal": null,
    "stdout": "/QSYS.LIB/QZDFMDB2.PGM",
    "stderr": ""
}

/: ls -p /QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh
{
    "code": 0,
    "signal": null,
    "stdout": "/QIBM/ProdData/IBMiDebugService/bin/startDebugService.sh",
    "stderr": ""
}

/: ls -p /QSYS.lib/ILEDITOR.lib/GE*.PGM
{
    "code": 2,
    "signal": null,
    "stdout": "",
    "stderr": "ls: cannot access '/QSYS.lib/ILEDITOR.lib/GE*.PGM': No such file or directory"
}

/: ls -p /QSYS.lib/ILEDITOR.lib/GE*.PGM
{
    "code": 2,
    "signal": null,
    "stdout": "",
    "stderr": "ls: cannot access '/QSYS.lib/ILEDITOR.lib/GE*.PGM': No such file or directory"
}

/: test -x /QOpenSys/QIBM/ProdData/JavaVM/jdk80/64bit/bin/java
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -x /QOpenSys/QIBM/ProdData/JavaVM/jdk11/64bit/bin/java
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -x /QOpensys/pkgs/lib/jvm/openjdk-11/bin/java
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -x /QOpenSys/QIBM/ProdData/JavaVM/jdk17/64bit/bin/java
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
SELECT * FROM QSYS2.ASP_INFO
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i')"
select SYSTEM_VALUE_NAME, CURRENT_NUMERIC_VALUE from QSYS2.SYSTEM_VALUE_INFO where SYSTEM_VALUE_NAME = 'QCCSID'
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

/: echo $PATH
{
    "code": 0,
    "signal": null,
    "stdout": "/QOpenSys/pkgs/bin:/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin:/QOpenSys/usr/bin:/usr/bin",
    "stderr": ""
}

/: test -e //.bashrc
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
liblist -d 
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: /QOpenSys/usr/bin/expr `/QOpenSys/usr/bin/getconf ARG_MAX` - `env|wc -c` - `env|wc -l` \* 4 - 2048
{
    "code": 0,
    "signal": null,
    "stdout": "4191772",
    "stderr": ""
}

/: rm -rf /tmp/code4itempO_WSe1iyFxmYSJgF57jTiW && mkdir -p /tmp/code4itempO_WSe1iyFxmYSJgF57jTiW
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: /QOpenSys/usr/bin/qsh
system  "RUNSQLSTM SRCSTMF('/tmp/code4itempO_WSe1iyFxmYSJgF57jTiW/getnewlibl.sql') COMMIT(*NONE) NAMING(*SQL)"
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: rm -rf /tmp/code4itempO_WSe1iyFxmYSJgF57jTiW
{
    "code": 0,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
select LONG_COMMENT from qsys2.sysroutines where routine_schema = 'ILEDITOR' and routine_name = 'GETMBRINFO'
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
CREATE TABLE QTEMP.O_09FVS AS (select job_name, local_port from qsys2.netstat_job_info where cast(local_port_name as VarChar(14) CCSID 37) = 'is-debug-ile' fetch first row only) WITH DATA;
Call QSYS2.QCMDEXC('CPYTOIMPF FROMFILE(QTEMP/O_09FVS *FIRST) TOSTMF(''/tmp/vscodetemp-O_EK3mFae7'') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM('','') DECPNT(*PERIOD)')
/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
CREATE TABLE QTEMP.O_W3TR6 AS (select job_name, local_port from qsys2.netstat_job_info j where job_name = (select job_name from qsys2.netstat_job_info j where local_port = 8005 and remote_address = '0.0.0.0' fetch first row only) and remote_address = '0.0.0.0') WITH DATA;
Call QSYS2.QCMDEXC('CPYTOIMPF FROMFILE(QTEMP/O_W3TR6 *FIRST) TOSTMF(''/tmp/vscodetemp-O_lMVgjQUj'') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM('','') DECPNT(*PERIOD)')
/: test -r /QIBM/ProdData/IBMiDebugService/package.json
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
CREATE TABLE QTEMP.O_P93HX AS (select job_name, local_port from qsys2.netstat_job_info where cast(local_port_name as VarChar(14) CCSID 37) = 'is-debug-ile' fetch first row only) WITH DATA;
Call QSYS2.QCMDEXC('CPYTOIMPF FROMFILE(QTEMP/O_P93HX *FIRST) TOSTMF(''/tmp/vscodetemp-O_29haEoIn'') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM('','') DECPNT(*PERIOD)')
/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
CREATE TABLE QTEMP.O_DTNA8 AS (select job_name, local_port from qsys2.netstat_job_info j where job_name = (select job_name from qsys2.netstat_job_info j where local_port = 8005 and remote_address = '0.0.0.0' fetch first row only) and remote_address = '0.0.0.0') WITH DATA;
Call QSYS2.QCMDEXC('CPYTOIMPF FROMFILE(QTEMP/O_DTNA8 *FIRST) TOSTMF(''/tmp/vscodetemp-O_SDDlpKt8'') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM('','') DECPNT(*PERIOD)')
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

/: LC_ALL=EN_US.UTF-8 system "call QSYS/QZDFMDB2 PARM('-d' '-i' '-t')"
CREATE TABLE QTEMP.O_SRBGT AS (select OS_VERSION concat '.' concat OS_RELEASE as VERSION from sysibmadm.env_sys_info) WITH DATA;
Call QSYS2.QCMDEXC('CPYTOIMPF FROMFILE(QTEMP/O_SRBGT *FIRST) TOSTMF(''/tmp/vscodetemp-O_Fv8qZHvq'') MBROPT(*REPLACE) STMFCCSID(1208) RCDDLM(*CRLF) DTAFMT(*DLM) RMVBLANK(*TRAILING) ADDCOLNAM(*SQL) FLDDLM('','') DECPNT(*PERIOD)')
/: test -r /QIBM/ProdData/IBMiDebugService/package.json
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -f /QIBM/UserData/IBMiDebugService/certs/debug_service.pfx
{
    "code": 0,
    "signal": null,
    "stdout": "DB2>\n                                                   PTXBEDEV-Traitement en Recette",
    "stderr": ""
}

{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -f /QIBM/UserData/IBMiDebugService/certs/debug_service.crt
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -f /QIBM/UserData/IBMiDebugService/.code4i.debug
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

/: test -f /QIBM/ProdData/IBMiDebugService/bin/certs/debug_service.pfx
{
    "code": 1,
    "signal": null,
    "stdout": "",
    "stderr": ""
}

**Reply by sebjulliand** [reply]
My assumption here is that there is this custom message that's returned by the DB2 CLI: `PTXBEDEV-Traitement en Recette`
There's i a good chance it's interfering with the extension's processes.
Do you know what is producing this output? (a trigger, an exit program?)

**Reply by cirikus** [reply]
I don't know...i need to check this ;)
